### DISCUSSION TITLE
Make validation `relaxed` or `strict` while syncing

### DISCUSSION THREAD

**Discussion by EclesioMeloJunior** [ChainSafe/gossamer#3176] [Ideas]
The validation that ensures progress in the Justifications was removed by https://github.com/ChainSafe/gossamer/pull/3167 but I would like to discuss another option which is: creating a flag to enable/disable this validation and the argument is:

- After a chat with Andre from Parity he suggested:

> if you are implementing such a validation then I'd suggest allowing to make it "relaxed", to deal with at least this case on Westend

- And he explains that Substrate doesn't implement such validation:

> we don't really look at the round numbers when validating justifications, at least not in the sense of comparing it to previous justifications

> whenever we validate justifications we only look at the round number to verify the signatures, we don't actually validate that the round number is > than what we had seen previously for another finalized block

So, I see that we can remove the validation as I propose in PR https://github.com/ChainSafe/gossamer/pull/3167 but I found it worth bringing up this discussion so we can argue about keeping the validation but enabling it for `Kusama/Polkadot` and disabling it for `Westend`

**Comment by jimjbrettj** [comment]
@EclesioMeloJunior I mean I think if we want to have the option to run in either mode I think this is a nice way to handle it. However, my question is do we want/need this functionality? If Substrate only used relaxed verification for Justifications, is there a reason we wouldn't just support that?

I guess what I am asking is this, for what reason do we have to give the option of running strict mode? Is there any benefit? If not my feeling is we just mimic substrate and do relaxed verification
